### PR TITLE
Fix registering DID/VC and IRMA login flow for new Nuts Node

### PR DESCRIPTION
**Pull request by reinkrul**
I apologize for my poor NodeJS skills.

### CODE REVIEW

File: api/browser/authentication.js [hunk 1]
```diff
@@ -2,56 +2,56 @@ const router = require('express').Router()
...
-      // TODO: don't depend on the irma-demo schemeManager here
-      const fullNameEntry = status.disclosed.find(el => el.identifier === 'irma-demo.gemeente.personalData.fullname')
-
-      // This is still experimental, and depends on the SSO feature to land in
-      // the Nuts node. When attribute not available, fall back on default user:
-      if (fullNameEntry)
-        req.session.user = fullNameEntry.rawvalue
-      else
-        req.session.user = req.session.user || config.organisation.user
+module.exports = function(organisation) {
```
Comment (Timendus) [inline api/browser/authentication.js:5]: Why did you change this structure to pass this particular organisation object down through functions? You could have put the DID on `config.organisation.did` or something similar, and have it available everywhere.

File: resources/nuts-node/auth.js [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
-  createJwtBearerToken: async (context) => call('createJwtBearerToken', null, context),
+  drawUpContract: async (organisationDID) => callExperimental('drawUpContract', null, {
+    type: 'BehandelaarLogin',
+    language: 'NL',
+    version: 'v3',
+    legalEntity: organisationDID,
+  }),
+  createLoginSession: async (contract) => callExperimental('createSignSession', null, {means: 'irma', payload: contract.message, params: {}}),
+  createSession: async (contract) => callV0('createSession', null, contract),
+  sessionRequestStatus: async (id) => callExperimental('getSignSessionStatus', id),
```
Comment (Timendus) [inline resources/nuts-node/auth.js:25]: Why do some methods use the experimental version of the API while others don't? Are both APIs available on different endpoints on the same server operating on the same data? Will this not cause issues in the future?

File: resources/nuts-node/vcr.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const config = require('../../util/config')
+const apiHelper = require('./open-api-helper')
+
+const call = apiHelper.call({
+    baseURL: `http://${config.nuts.node}`,
+    definition: 'https://raw.githubusercontent.com/nuts-foundation/nuts-node/master/docs/_static/vcr/v1.yaml'
+});
```
Comment (Timendus) [inline resources/nuts-node/vcr.js:7]: >hoewel ik moet zeggen dat ik aangenaam verrast was door de eenvoud waarmee je OpenAPI calls doet a.d.h.v. een YAML spec

Ik was er ook wel happy mee 😄

File: api/events/consent.js [hunk 4]
```diff
@@ -1,226 +1,230 @@
...
-    } catch (e) {
-      orgs.push({
-        identifier: urn,
-        name: 'Could not find organisation: ' + e
-      })
-    }
-  }
-  return orgs
-}
+// The functionality this file contains is deprecated because it's being reworked in `nuts-node`.
```
Comment (Timendus) [inline api/events/consent.js:1]: Then why touch this file at all in this PR? Did it break things?

**Review by Timendus — COMMENTED** [review]
I don't think your skills are that poor 😜 Nice work. Can you help me to test this locally on my machine? Do I really just check out `https://github.com/nuts-foundation/nuts-node/tree/master/development` and run `make`..? 😄

File: api/browser/authentication.js [hunk 5]
```diff
@@ -2,56 +2,56 @@ const router = require('express').Router()
...
+      const status = await auth.sessionRequestStatus(req.session.nuts_session)
+      if (status.status === 'DONE') {
+        const verifyResponse = await auth.verifySignature(status.verifiablePresentation)
+        req.session.nuts_auth_token = status.nuts_auth_token
+
+        const attrs = verifyResponse.issuerAttributes;
+        // Extract the user's full name from the disclosed attributes.
+        // TODO: don't depend on the irma-demo schemeManager here
+        req.session.user = [
+          attrs["gemeente.personalData.initials"],
```
Comment (stevenvegt) [inline api/browser/authentication.js:28]: Not quite sure, but perhaps introspect token api can help you with extracting these values?

File: resources/nuts-node/auth.js (hunk 2 above)
Comment (stevenvegt) [reply resources/nuts-node/auth.js:25]: This will be merged into one v1 api soon

File: README.md [hunk 6]
```diff
@@ -56,35 +55,25 @@ records in the config files.
...
 ```yaml
 auth:
-  irmaSchemeManager: irma-demo
+  irma:
+    schememanager: irma-demo
 ```
 
-You will need to restart your Nuts nodes to enable these changes.
-
 #### NATS events
```
[the remark this answers is not in the document]
Comment (Timendus) [reply README.md:77]: What's the new event bus?

File: resources/nuts-node/auth.js (hunk 2 above)
Comment (Timendus) [reply resources/nuts-node/auth.js:25]: 👍🏻

Comment (reinkrul) [reply resources/nuts-node/auth.js]: Fixed

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (reinkrul) [reply README.md:77]: Nothing yet.

File: api/events/consent.js (hunk 4 above)
Comment (reinkrul) [reply api/events/consent.js:1]: Tested, didn't break anything. Reverted the change.

**Comment by Timendus** [comment]
Je alter-ego @reinkrul? ;)

**Comment by reinkrul** [comment]
Further improvements will be done when revamping the project to make it integrate with the regsitry admin demo.
